chaincode/ledger.go

```diff
@@ -215,16 +216,22 @@ type ComputePlan struct {
 	Tag                     string               `json:"tag"`
 	TesttupleKeys           []string             `json:"testtuple_keys"`
 	TraintupleKeys          []string             `json:"traintuple_keys"`
+	Workers                 []string             `json:"workers"`
 }
 
 // ComputePlanState is the ledger's representation of the compute plan state.
 // To minimize the size of every compute plan, update its state record under another
 // key in the ledger. It will reduce the growing rate of the blockchain size.
 type ComputePlanState struct {
-	DoneCount               int      `json:"done_count"`
+	Status string `json:"status"`
+}
+
+// ComputePlanWorkerState contains state information for a given
+// compute plan and worker
+type ComputePlanWorkerState struct {
 	IntermediaryModelsInUse []string `json:"intermediary_models_in_use"`
-	Status                  string   `json:"status"`
```

Comment:
this looks as a breaking change. maybe we could keep it for now (or deprecated first)? is the backend updated also?

Reply:
There's no update necessary in the backend: to the backend the change is transparent.

Re: breaking. I ran a test. I started two compute plans with the current `master` version of the chaincode. 

![image](https://user-images.githubusercontent.com/168382/107063989-066ef100-67a9-11eb-9fb5-6cf5d3d5a8d8.png)


Once the CPs were done I upgraded the chaincode to this branch. This is what I see in `substra list compute_plan`

![image](https://user-images.githubusercontent.com/168382/107064054-15ee3a00-67a9-11eb-81c2-122f1f03d9db.png)


So we lose the counts, but we can safely "get" compute plans without an error. 

I haven't tested adding a new tuple to the existing CP. In that case we would have undetermined behavior. Handling full backwards compatibility would be much more work, as it would involve writing a full migration.

Reply:
ok, let's keep things simple for now and tackle the migration process in another track.

We need at least make sure that it is mentioned in the next release and update the backend to handle the status properly + update compatibility table as usual.